File: backend/src/main/java/de/neuefische/backend/model/Capstone.java
package de.neuefische.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Document("capstones")
public class Capstone {
    @Id
    private String id;
    private String githubApiUrl;
    private String qualityBadgeUrl;
    private String coverageBadgeUrl;

    private String studentName;
    private String url;
    private LocalDate updatedAt;
    private int allCommits;
    private int mainCommits;
    private int allPulls;
    private int openPulls;
}

File: frontend/src/models/Capstone.ts
export default interface Capstone {
  id: string
  githubApiUrl: string
  studentName: string
  url: string

  updatedDefaultAt: string
  updatedAt: string
  allCommits: number
  mainCommits: number
  allPulls: number
  openPulls: number

  qualityBadgeUrl?: string
  coverageBadgeUrl?: string

  workflowBadgeUrl?: string
}

export type CapstoneRequest = {
  githubRepoUrl: string
}
